src/index.ts

```diff
@@ -1,123 +1,106 @@
-	}
-
-	return config
+async function chooseConfig(): Promise<AvailableConfig> {
+	return prompts({
+		type: 'select',
+		name: 'config',
+		message: "Which ESLint configuration do you want to install?",
+		choices: availableConfigs.map((config) => ({
+			title: config,
+			value: config
+		}))
+	}).then(({config}) => config)
 }
 
 function createESLintConfig(whichConfig: AvailableConfig) {
-	const { config } = configs[whichConfig]
-
 	const dataAsString = JSON.stringify(
-		config,
+		{
+      extends: `@monogram/eslint-config/${whichConfig}`
+    },
 		null,
-		2 // TODO: get config from prettier
+		2 
 	)
 
 	writeFileSync(`${process.cwd()}/${ESLINT_FILENAME}`, `module.exports = ${dataAsString}`)
 
 	console.log(`✅ ESLint configuration file created at ${process.cwd()}/${ESLINT_FILENAME}`)
 }
 
-async function choosePackageManager() {
-	let packageManager = findPackageManager()
-
-	if (packageManager) {
-		console.log(`\n📦 An existing ${packageManager} installation was found`)
-	} else {
-		const packageManagersOptions = packageManagers.join(', ')
-
-		const chosenManager = await question(
-			`Which package manager should be used? \n[${packageManagersOptions}] `
-		)
-
-		if (packageManagersOptions.includes(chosenManager as PackageManager)) {
-			packageManager = chosenManager as PackageManager
-		}
-	}
-
-	return packageManager
+async function choosePackageManager(): Promise<PackageManager> {
+	return prompts({
+		type: 'select',
+		name: 'packageManager',
+		message: "Which ESLint configuration do you want to install?",
+		choices: packageManagers.map((pm) => ({
+			title: pm,
+			value: pm
+		})),
+		initial: findPackageManager()
+	}).then(({packageManager}) => packageManager)
 }
 
-async function installDependencies(config: AvailableConfig, packageManager: PackageManager) {
-	const installPrefix = installPrefixes[packageManager]
+const installPrefixes = {
+	yarn: 'yarn add -D',
+	pnpm: 'pnpm add -D',
+	npm: 'npm i -D'
+} as const
 
-	const { dependencies } = configs[config]
-	const dependenciesAsString = dependencies.join(' ')
+async function installDependency(packageManager: PackageManager) {
+	const installPrefix = installPrefixes[packageManager]
 
-	const installCommand = `${installPrefix} ${dependenciesAsString}`
+	const installCommand = `${installPrefix} eslint @monogram/eslint-config`
 
 	console.log(`📦 Installing dependencies...`)
-	console.log(`${installCommand}\n`)
-
-	await exec(
-		installCommand,
-		// eslint-disable-next-line @typescript-eslint/ban-ts-comment
-		// @ts-ignore
-		(stdout: string, stderr: string) => {
-			if (stderr) {
-				console.error(stderr)
-			} else {
-				console.log(stdout)
-			}
-
-			Promise.resolve({ stdout, stderr })
-		}
-	)
+
+  try {
+    execSync(installCommand)
+  } catch (error) {
+    console.error(error)
+  }
 }
+
+const lockFiles = {
+	yarn: `${process.cwd()}/yarn.lock`,
+	pnpm: `${process.cwd()}/pnpm-lock.yaml`,
+	npm: `${process.cwd()}/package-lock.json`
+}
```

Comment:
this should follow the package manager types
```suggestion
const lockFiles: Record<PackageManager, string> = {
	yarn: `${process.cwd()}/yarn.lock`,
	pnpm: `${process.cwd()}/pnpm-lock.yaml`,
	npm: `${process.cwd()}/package-lock.json`
}
```

Reply:
The type gets inferred automatically. I prefer inferred, but if you want it explicitly typed we can do that! Another option is to do inferred but add `as const` to make sure its read-only and get even better type-inference.

Reply:
Even if the type is inferred it does not know about the 'supported package managers' we have, so one could do:

```ts
const lockFiles = {
	yarn: `${process.cwd()}/yarn.lock`,
	pnpm: `${process.cwd()}/pnpm-lock.yaml`,
	npm: `${process.cwd()}/package-lock.json`,
	invalid: 'lock'
}
```

without getting type errors.

I see it was not typed before(my bad) but we can make this improvement now as we're touching this code :)

Reply:
I see! Sounds good